docs/eove-his-interface.md

```diff
@@ -0,0 +1,765 @@
+- read data from EO150 station
+- respond to EO150 station pings to keep the link alive
+
+## Data format
+
+Data exchanged over USB link are JSON strings separated by a new line character and encoded in UTF-8 bytes.
+
+Example of such string before UTF-8 encoding:
+
+```
+{"type":"DO_SOMETHING"}\n
+```
+
+Client can either send or receive such messages.
+
+## Plumbing
+
+### Accessory mode
+
+As a prerequisite, USB cable must be plugged on micro USB port.
+
+Trigger Accessory Mode according to https://source.android.com/devices/accessories/aoa.
+
+Reference implementation in [accessoryModeConfigurator.ts](https://github.com/eove/his-API/blob/master/node-client/lib/usb/accessoryModeConfigurator.ts).
+
+In summary steps are:
+
+- find Android device based on usual Android USB ids
+- send a control transfer in request to validate version code
+- send multiple control transfer out requests with accessory information
+- send a control transfer out request to put Android device in accessory mode
+
+Note that an Android device stays in accessory mode, there is no command to put it in default mode back.
+
+The device leaves this mode automatically when:
+
+- USB cable is unplugged
+- USB is reset on accessory side
+- Android device reboots
+
+Also note that only one accessory can be connected to an Android device.
+This means you cannot connect two clients to the same server.
+
+### Communication ping/pong mechanism
+
+To keep the connection alive, pings are emitted periodicaly by the server (typically every 8s).
+The client shall reply with special [pongs messages](#pong-command) before the timeout (typically 5s).
+
+Client reads:
+
+```json
+{
+  "type": "PING"
+}
+```
+
+Client writes:
+
+```json
+{
+  "type": "PONG"
+}
+```
+
+When client fails to send a pong message, server will consider client as disconnected.
+This means that client will have to enable [high level communication](#high-level-communication) again.
+
+### Data format
+
+When decoded and delimited, messages are JSON-valid strings.
+In other words every used types are compatible with JSON (number, string, array, etc.).
+
+Dates (with time component) are converted to epoch with a millisecond precision.
+
+## Client messages
+
+Client can send message to server and these messages are mostly commands.
+This means message type will be an order like `DO_SOMETHING` or `GET_ME_SOMETHING`.
+
+Commands will often receive a positive response message like `DO_SOMETHING_SUCCEEDED`.
+
+`DO_SOMETHING` or `GET_ME_SOMETHING` are not valid commands but are used in several examples because they are concise enough.
+
+A message has the following simplified type:
+
+```ts
+interface Message {
+  type: string;
+  reference?: string;
+  payload?: any;
+}
+```
+
+### High level communication
+
+By default server will ignore any message coming from client because high level communication is disabled.
+Client should send a [START_COMMUNICATION](#startcommunication-command) command to enable it.
```

Comment:
I would say: communication is not considered as "started"... to avoid "disabled" which is no the same semantic meaning.